pkg/cmd/openshift-tests/disruption/poll-service/poll_service.go

```diff
@@ -0,0 +1,65 @@
+package poll_service
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"net"
+	"time"
+
+	"github.com/openshift/origin/pkg/monitor/backenddisruption"
+	"github.com/openshift/origin/pkg/monitor/monitorapi"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/klog/v2"
+)
+
+type ServicePoller struct {
+	Address     string
+	Port        string
+	serviceName string
+	recorder    monitorapi.RecorderWriter
+	outFile     io.Writer
+}
+
+func NewServicePoller(clusterIP, port string, outFile io.Writer) *ServicePoller {
+	return &ServicePoller{
+		Address: clusterIP,
+		Port:    port,
+		outFile: outFile,
+	}
+}
+
+// Run starts polling and blocks until stopCh is closed.
+func (p *ServicePoller) Run(ctx context.Context, finishedCleanup chan struct{}) {
+	defer utilruntime.HandleCrash()
+	defer close(finishedCleanup)
+
+	logger := klog.FromContext(ctx)
+	logger.Info("Starting Polling Service")
+	defer logger.Info("Shutting down Service Poller")
+
+	go wait.UntilWithContext(ctx, p.runServicePoller, time.Second)
+	//go p.runServicePoller(ctx)
+
+	// do a thing
+
+	<-ctx.Done()
+
+}
+
+func (p *ServicePoller) runServicePoller(ctx context.Context) {
```

Comment:
I think you still want logic like https://github.com/openshift/origin/blob/master/pkg/cmd/openshift-tests/disruption/watch-endpointslice/endpointslice_controller.go#L117-L127 to cleanly stop your samplers to have them write output.